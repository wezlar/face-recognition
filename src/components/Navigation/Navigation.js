/** @jsx jsx */
import { jsx, css } from '@emotion/core';
import PropTypes from 'prop-types';
import NavigationLink from '../NavigationLink/NavigationLink';

const style = css`
  margin-top: 0;
  padding-left: 0;
  background-color: #3939a0;

  ul {
    list-style: none;
    display: flex;
    padding: 0;
  }
`;

export const Navigation = (props = {}) => {
  const { links = [] } = props;
  
  return (
    <div css={style}>
      <ul>
      {links.map((link, index) => 
        <NavigationLink 
          key={index} 
          label={link.label} 
          url={link.url} />
      )}
      </ul>
    </div>
  );
};

Navigation.propTypes = {
  links: PropTypes.array,
};

export default Navigation;
