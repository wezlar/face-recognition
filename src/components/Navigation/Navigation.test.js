// __tests__/fetch.js
import React from 'react';
import serializer from 'jest-emotion'
import {
  render,
  cleanup,
} from '@testing-library/react';
// this adds custom jest matchers from jest-dom
import renderWithRouter from '../../../testUtils/renderWithRouter';
import Navigation from './Navigation';

afterEach(cleanup);

const mockLinks = [
  {
    label: 'Home Page',
    url: '/',
  },
  {
    label: 'Test Page 1',
    url: '/test-page-1',
  },
  {
    label: 'Test Page 2',
    url: '/test-page-2',
  },
];

describe(`Test Navigation`, () => {
  test(`Test snapshot`, () => {
    // Arrange
    const { container, debug } = renderWithRouter(
      <Navigation links={mockLinks} />,
    );

    // snapshots work great with regular DOM nodes!
    expect(container.firstChild).toMatchSnapshot();
  });


  test(`Test snapshot with empty props`, () => {
    // Arrange
    const { container, debug } = renderWithRouter(Navigation());

    // snapshots work great with regular DOM nodes!
    expect(container.firstChild).toMatchSnapshot();
  });
});