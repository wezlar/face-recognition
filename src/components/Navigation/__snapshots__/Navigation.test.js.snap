// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navigation Test snapshot 1`] = `
.emotion-3 {
  margin-top: 0;
  padding-left: 0;
  background-color: #3939a0;
}

.emotion-3 ul {
  margin-top: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.emotion-0 {
  background-color: #3939a0;
  display: block;
  padding: 20px 30px;
  color: #fff;
}

.emotion-0:hover {
  background-color: #272781;
}

.emotion-3 {
  margin-top: 0;
  padding-left: 0;
  background-color: #3939a0;
}

.emotion-3 ul {
  margin-top: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.emotion-0 {
  background-color: #3939a0;
  display: block;
  padding: 20px 30px;
  color: #fff;
}

.emotion-0:hover {
  background-color: #272781;
}

<div
  class="emotion-3"
>
  <ul>
    <li>
      <a
        class="emotion-0"
        href="/"
      >
        Home Page
      </a>
    </li>
    <li>
      <a
        class="emotion-0"
        href="/test-page-1"
      >
        Test Page 1
      </a>
    </li>
    <li>
      <a
        class="emotion-0"
        href="/test-page-2"
      >
        Test Page 2
      </a>
    </li>
  </ul>
</div>
`;

exports[`Test Navigation Test snapshot with empty props 1`] = `
.emotion-0 {
  margin-top: 0;
  padding-left: 0;
  background-color: #3939a0;
}

.emotion-0 ul {
  margin-top: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.emotion-0 {
  margin-top: 0;
  padding-left: 0;
  background-color: #3939a0;
}

.emotion-0 ul {
  margin-top: 0;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

<div
  class="emotion-0"
>
  <ul />
</div>
`;
